modules/internal/DefaultRectangular.chpl

```diff
@@ -1343,6 +1343,18 @@ module DefaultRectangular {
                                                    BoundedRangeType.bounded,
                                                    stridable)) {
 
+      // check to see whether this realloc is actually changing the
+      // bounds of the array
+      var actuallyResizing = false;
+      for i in 1..rank {
```

Comment:
Could this utilize indices?

```chpl
for i in bounds.indices {
  ..
}
```

Reply:
That's a nice idea, yes it should be able to.